### PR TITLE
Remove all of Application Cache

### LINKED ISSUE #40: Remove appcache from future HTML specs
It is something that doesn't do what people expect, and while it is used today in combination with web storage and other techniques, we expect it to be rendered truly obsolete by Service Workers.

There is an HTML5 recommendation, showing people how to implement `appcache`, if they want to make 2015-era browsers.

[Firefox has deprecated, and plans to remove it](https://www.fxsitecompat.com/en-CA/docs/2016/application-cache-support-will-be-removed/) in favour of service workers.

At some point, HTML specs should stop recommending its use. Have we reached that point yet?

### PR DESCRIPTION
**Pull request by arronei**
- Remove all application cache definitions
- Remove `manifest` attribute

Fix #40, #167

We should probably create a Note document for this removed section and information.

### CODE REVIEW

File: sections/events.include [hunk 1]
```diff
@@ -256,4 +256,4 @@
     </tbody>
   </table>
 
-  See also <a href="#media-elements-event-summary">media element events</a>, <a href="#application-cache-api">application cache events</a>, and <a>drag-and-drop events</a>.
\ No newline at end of file
+  See also <a href="#media-elements-event-summary">media element events</a>, and <a>drag-and-drop events</a>.
```
Comment (travisleithead) [inline sections/events.include]: Comma unnecessary.

File: sections/semantics-root.include [hunk 2]
```diff
@@ -38,35 +37,9 @@
...
 
-  Authors are encouraged to specify a <{global/lang}> attribute on the root <{html}>
-  element, giving the document's language. This aids speech synthesis tools to determine what
-  pronunciations to use, translation tools to determine what rules to use, and so
-  forth.
-
-  <p class="note">
-    It is recommended to keep the usage of attributes and their values defined on the <{html}>
-    element to a minimum to allow for proper detection of the <a>character encoding declaration</a>
-    within the first 1024 bytes.
```
Comment (travisleithead) [inline sections/semantics-root.include:49]: I think the above paragraph should not be removed; this isn't about app-cache, but character encoding, and is still good advice.

**Comment by travisleithead** [comment]
Just a couple of comments to address. Otherwise LGTM.

**Comment by travisleithead** [comment]
Looks Great!

**Comment by annevk** [comment]
How does the WG justify this given the four implementations?

**Comment by adanilo** [comment]
HTML 5.0 specifies the behavior of Application Cache and 5.0 compliant user agents are free to implement it with the existing reference Recommendation.
As current browsers are implementing Service Worker as a more flexible solution to offline caching and resource management, the requirement for Application Cache remaining in the spec. is at best poor advice for developers, and so is being removed to help keep the spec. in line with best practice for browsers moving forward.
If somehow, Application Cache becomes more useful as an offline caching mechanism in ways that have not been anticipated, it may be re-integrated into a future version.
As it stands, the number of implementations in existence fails to validate the design or usefulness of the feature.

**Comment by annevk** [comment]
Seems anti-competitive since as other features evolve AppCache will be affected too. If you don't define those interactions new user agents will have a harder time entering the market.

**Comment by adanilo** [comment]
Would you like to write a new user agent?

**Comment by annevk** [comment]
I'm not writing one myself, but what about, e.g., Servo?

**Comment by annevk** [comment]
Also, how does that question address the concern in any way?

**Comment by adanilo** [comment]
Servo looks great, I've personally been watching it's progress. Your comment about 'anti-competitive' seems rather trite and childish and misses the point. Could you, in all honesty advise a web developer to use Application Cache? Or are you just arguing for the sake of it? I don't actually see what the concern is. e.g. in W3C specs, we had multiple examples of SVG fonts implemented in Adobe SVG, Batik, Bitflash, Ikivo, Abbra, etc. which amounts to 5 implementations, yet none of the 4 major browsers adopted them and they are to deprecated in SVG 2.0. Similarly, Application Cache has a number of implementations yet its usefulness is questionable at best, and so, will be removed for the next version of the W3C spec to reflect what seems to be the developer and browser developer community best practice moving forward. This does not mean removal from existing user agents, just that new exciting code-bases like Servo should not bother implementing broken (IMHO) designs.

**Comment by annevk** [comment]
Whether web developers should or should not use something is immaterial to whether Servo needs to implement it. The W3C's copy of the HTML Standard defines many things that developers should not use, yet Servo needs to implement. Your analogy doesn't hold since implementations outside browsers have no impact on what is important for implementing a browser. Anyway, I guess it doesn't matter since browsers don't implement from the copy anyway, but it's definitely anti-competitive.

**Comment by chaals** [comment]
The chairs had agreed that this potential change would require a formal consensus of the Working Group, and that hasn't happened. Travis, please revert the change for now, and I will work on the CfC.

**Comment by chaals** [comment]
@Annevk your claim that this is somehow anti-competitive behaviour is actually a pretty serious assertion, despite the fact that you appear to thoroughly undermine it in the immediately preceeding sentence.

It would be appropriate to carefully substantiate the argument, or to withdraw the assertion. It is not appropriate to make such allegations without a clearly substantiated case, written in a professional manner.

**Comment by annevk** [comment]
If your aim is to have a document that implementers can use as a basis to compete on and you omit a crucial part of what all the existing implementers have and how that part evolves along the way, you have created a situation where a new entrant cannot effectively compete and has to resort to reverse engineering.

**Comment by rniwa** [comment]
I agree with @annevk here.  I don't think we can remove AppCache for compatibility reasons.  We need to keep it as an deprecated/obsolete feature at least.

**Comment by travisleithead** [comment]
In previous discussions I've heard talk of moving AppCache to a note. The arguments against that were "why bother" since it's published in HTML5.0 Rec.

The new concern I'm hearing is that as HTML's algorithms are clarified or new ones added in the future, AppCache will be forgotten, will become stale, unmaintainable, and not implementable.

On the flip side, all browsers have this code running in their UA's, and none of them (to my knowledge) care about improving or enhancing it. The only reason they don't rip it out, is due to web compatibility concerns. I believe there is even an expectation of removing the feature at some future time as the web has moved on and no longer needs it.

So... I wonder if there is really a concern about keeping it in the spec for the purpose of keeping it up to date, or said another way, would anyone care if AppCache stopped being maintained? Implementations will have to care, but specs aren't implementations...

**Comment by plehegar** [comment]
the CfC is out:
  https://lists.w3.org/Archives/Public/public-html/2016May/0017.html

**Pull request merged**